libs/dh/admin/data-access-api/src/lib/dh-custom-data-source.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DataSource } from '@angular/cdk/collections';
+import { inject } from '@angular/core';
+
+import { UserOverviewItemDto } from '@energinet-datahub/dh/shared/domain';
+
+import { DhAdminUserManagementDataAccessApiStore } from './dh-admin-user-management-data-access-api.store';
+
+export class DhCustomDataSource implements DataSource<UserOverviewItemDto> {
+  private readonly store = inject(DhAdminUserManagementDataAccessApiStore);
+
+  connect() {
+    return this.store.users$;
+  }
+
+  disconnect(): void {
+    // Intentionally left empty.
+  }
+}
```

Comment:
suggestion: perhaps follow the convention from the material library by adding a paginator accessor and then control pagination from the data source instead?

Reply:
Do you have an example I can look at? I'm not sure I can follow the suggestion.

Reply:
https://github.com/angular/components/blob/main/src/material/table/table-data-source.ts#L131

Reply:
Yeah, the implementation obviously needs to be different. I was imagining it would work much the same as you are doing now, but instead of subscribing to the PageEvent in your component, you would subscribe to it in the DataSource - and then just update the page metadata from within the DataSource.

Reply:
Probably not much to gain from that right now, but it would make it easier to transition into a more generic and reusable DataSource that could be used in other areas as well. I imagine server side pagination is going to be needed in multiple places in the future.

Reply:
As it is right now, the data source is destroyed and recreated every time a new page data has loaded. That is because the table and paginator are in a component controlled by a `isLoading` flag used to show a spinner. So the entire page content is removed, a spinner is shown, and the recreated again once a request completes.

So in order for this to work, the loading spinner probably needs to be all the way next to the table, so only the table is hidden when fetching data. This way the paginator will always be visible on the page. That's a detail I haven't worked out yet.

Reply:
But you're right that server side pagination is going to be needed in other places as well. So that's a use case that must be taken into consideration.

Reply:
That makes it harder for sure 😄 Doesn’t sound right to rebuild everything when changing pages. Perhaps you should have two separate loading states, one for loading the table initially and another for when changing the page?

Reply:
Two loading states is probably is going to be the best solution.

Reply:
Looked quickly at a two spinner solution, and it doesn't seem as promising. Hiding the table behind an `ngIf` in any way will mean that the related data source will be destroyed and recreated again. Couldn't find a way to show a spinner from within a table and still keep the table's elements (like column names. etc.). So I'm back to the drawing board.